aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -698,15 +716,16 @@ Tensor scaled_dot_product_attention(
           query_, key, value, attn_mask_, dropout_p, is_causal, scale, enable_gqa);
   }
   sdp::SDPBackend backend = static_cast<sdp::SDPBackend>(choice_int);
-  std::optional<Tensor> attn_mask = convert_boolean_attn_mask(attn_mask_, query_.dtype());
   switch (backend) {
     case sdp::SDPBackend::cudnn_attention: {
+      std::optional<Tensor> attn_mask = convert_boolean_attn_mask_cudnn(attn_mask_, query_.dtype());
```

Comment:
Doing this for the other backends won't cause numeric / efficiency issues, will it? Wouldn't updating the CUDNN tests be easier?

Reply:
Other backends should be unchanged as they are using the original conversion function. Even for cuDNN this isn't likely to affect numerics. Changing the tests is likely less maintainable as they would need to be changed every time a new workload is dispatched to cuDNN rather then mem-eff attention